ci/conda_env_cpp.txt

```diff
@@ -31,16 +31,16 @@ gflags
 glog
 gmock>=1.10.0
 google-cloud-cpp>=1.34.0
-grpc-cpp<=1.50.1
 gtest>=1.10.0
+libgrpc
 libprotobuf
 libutf8proc
 lz4-c
 make
 meson
 ninja
 nodejs
-orc
+orc<=2.0.3
```

Comment:
Once grpc is unpinned it is unfortunately required to pin orc to an older version. This traces back to an issue discovered here:

https://github.com/apache/arrow/pull/46906#issuecomment-3286997441

In more recent versions of orc, the library distributes libraries that interfere with CMake's dependency finder, and in a way that falsely reports many of the compression libraries to be missing